File: python/easy/1480_running_sum_of_1d_array.py
from typing import List


class Solution:
    # 52ms, 
    def runningSum(self, nums:List[int]) -> List[int]:
        return [ sum(nums[:i+1]) for i in range(len(nums)) ]
    
    # 
    def runningSum(self, nums:List[int]) -> List[int]:
        ans: List[int] = []
        temp = 0
        for i in nums:
            temp += i
            ans.append(temp)
        return ans

if __name__ == "__main__":
    so = Solution()
    print(so.runningSum(nums=[1, 2, 3, 4]))

File: c/easy/1480_Running_Sum_of_1d_Array.c
#include <stdio.h>
#include <stdlib.h>

// 12 ms, faster than 7.13% of C online submissions for Running Sum of 1d Array.
// 12 ms, faster than 7.13% of C online submissions for Running Sum of 1d Array.
int* runningSum(int* nums, int numsSize, int* returnSize){
    int cumsum = 0;
    int *result = (int *) malloc(numsSize * sizeof(int));
    * returnSize = numsSize;
    for (int i = 0; i < numsSize; i++){
        cumsum += nums[i];
        result[i] = cumsum;
    }
    return result;
}


int main(void){
    int nums[] = {3,1,2,10,1};
    int size;
    int i;
    int * results;

    results = runningSum(nums, sizeof(nums)/sizeof(int), &size);
    for (i = 0; i<sizeof(nums)/sizeof(int); i++){
        printf("  [%4d]  ",results[i] );
    }
    printf("\n");

    return 0;
}